### PR TITLE
Implements qml-import paths

### PR DESCRIPTION
**Pull request by JorisGoosen**
- Implements https://github.com/jasp-stats/INTERNAL-jasp/issues/1188
- just add a `inst/qmldir` with a module definition

### CODE REVIEW

File: Desktop/modules/dynamicmodules.cpp [hunk 1]
```diff
@@ -57,7 +58,7 @@ DynamicModules::DynamicModules(QObject *parent) : QObject(parent)
 
 DynamicModules::~DynamicModules()
 {
-	for(auto dynamic : _modules)
+	for(const auto & dynamic : _modules)
```
Comment (boutinb) [inline Desktop/modules/dynamicmodules.cpp]: A DynamicModule is a QObject with parent this DynamicModules QObject. So why do you have to delete it. It's even dangerous.

File: Desktop/modules/dynamicmodules.cpp [hunk 2]
```diff
@@ -167,8 +168,11 @@ bool DynamicModules::initializeModule(DynamicModule * module)
 			emit dynamicModuleChanged(module);
 			emit loadModuleTranslationFile(module);
 			restartEngines();
+
```
Comment (boutinb) [inline Desktop/modules/dynamicmodules.cpp:169]: Why this restartEngines does not have an emit. I know it does not do anything, but it's nice to see that you are using a signal.

File: Desktop/modules/dynamicmodules.cpp [hunk 3]
```diff
@@ -223,9 +227,24 @@ void DynamicModules::registerForInstallingModPkg(const std::string & moduleName)
...
+QStringList DynamicModules::importPaths() const
+{
+	QStringList allImportPaths;
+
+	for(const auto & nameMod : _modules)
+	{
+		QDir moduleFolder(tq(nameMod.second->moduleInstFolder()));
+
+		if(moduleFolder.exists("qmldir")) //So the inst contained a qmldir, and we give the library as an importpath
+			allImportPaths.append(nameMod.second->moduleRLibrary());
```
Comment (boutinb) [inline Desktop/modules/dynamicmodules.cpp:242]: the name moduleRLibrary is a bit awkward for the absolute path of the module folder.

**Review by boutinb — CHANGES_REQUESTED** [review]

File: Desktop/modules/dynamicmodules.cpp (hunk 1 above)
Comment (JorisGoosen) [reply Desktop/modules/dynamicmodules.cpp]: That is a good point!

Comment (JorisGoosen) [reply Desktop/modules/dynamicmodules.cpp]: Remnant from before, when they werent qobjects yet!
I just made the code look a bit better but didnt think about it.

File: Desktop/modules/dynamicmodules.cpp (hunk 2 above)
Comment (JorisGoosen) [reply Desktop/modules/dynamicmodules.cpp:169]: Yeah, I think they maybe weren't signals to begin with. I'll check em.

File: Desktop/modules/dynamicmodules.cpp (hunk 3 above)
Comment (JorisGoosen) [reply Desktop/modules/dynamicmodules.cpp:242]: Well, it isn't the absolute path of the module folder, it is the path to the r-library that contains the installed package that is the module folder. And it is particular for the module in question.

**Comment by JorisGoosen** [comment]
Ok changes applied!

**Review by boutinb — APPROVED** [review]

**Pull request merged**